### PR TITLE
CUMULUS-3243:Updated granule delete logic

### PR DESCRIPTION
**Pull request by jennyhliu**
**Summary:** Summary of changes

Addresses [CUMULUS-3243: Unable to delete granule from Postgres when it is missing in DynamoDB](https://bugs.earthdata.nasa.gov/browse/CUMULUS-3243)

## Changes

* Updated granule delete logic to delete granule which is not in DynamoDB
* Updated granule unpublish logic to handle granule which is not in DynamoDB and/or CMR

## PR Checklist

- [x] Update CHANGELOG
- [x] Unit tests
- [x] Ad-hoc testing - Deploy changes and test manually
- [ ] Integration tests

## Test
Unit tests are added.  I also tested manually:
1. Ingest and publish a granule
2. delete the granule from DynamoDB and Elasticsearch
3. update the cmr_link to 'https://cmr.uat.earthdata.nasa.gov/search/concepts/G123456-CUMULUS.echo10' which is not in CMR.
4. bulk delete this granule, e.g.
curl -X POST
  https://example.com/granules/bulkDelete --header 'Authorization: Bearer ReplaceWithTheToken' --header 'Content-Type: application/json' --data '{"ids": ["MOD09GQ.A2016358.h13v04.006.2016360104606"], "forceRemoveFromCmr": true}'

### CODE REVIEW

File: example/lambdas/python-reference-task/package.sh [hunk 1]
```diff
@@ -20,7 +20,7 @@ cp ./*.py ./dist/
 
 cd ./dist || exit 1
 
-find . -type f -print0 | xargs -0 node ../../../../bin/zip.js lambda.zip $(ls | grep -v lambda.zip)
+node ../../../../bin/zip.js lambda.zip $(ls | grep -v lambda.zip)
```
Comment (jennyhliu) [inline example/lambdas/python-reference-task/package.sh:23]: The change is already in master branch

File: packages/api/models/granules.js [hunk 2]
```diff
@@ -482,6 +483,19 @@ class Granule extends Manager {
     }
     return executionDescription;
   }
+
```
Comment (jennyhliu) [inline packages/api/models/granules.js:486]: When granule is not in dynamoDB, unpublish has error:
2023-04-18T15:46:33.756Z 489c96c9-fec7-411c-84b4-eac0df058c41 ERROR
{ "error": { "name": "RecordDoesNotExist", "message": "No record found for {\"granuleId\":\"MOD09GQ.A7614271.S4rfYb.006.4748017559580\"} in jl-tf-GranulesTable", "stack": [ "RecordDoesNotExist: No record found for {\"granuleId\":\"MOD09GQ.A7614271.S4rfYb.006.4748017559580\"} in jl-tf-GranulesTable", " at Object.get (/var/task/webpack:/src/DynamoDb.ts:63:9)", " at processTicksAndRejections (node:internal/process/task_queues:96:5)", " at Granule.get (/var/task/webpack:/models/base.js:228:12)", " at Granule.get (/var/task/webpack:/models/granules.js:88:29)", " at Granule.update (/var/task/webpack:/models/base.js:372:25)", " at /var/task/webpack:/lib/granule-remove-from-cmr.js:95:29" ] }, "level": "error", "message": "No record found for {\"granuleId\":\"MOD09GQ.A7614271.S4rfYb.006.4748017559580\"} in jl-tf-GranulesTable", "sender": "@cumulus/api", "timestamp": "2023-04-18T15:46:33.756Z" }

File: packages/api/src/lib/granule-delete.ts [hunk 3]
```diff
@@ -145,7 +147,7 @@ const deleteGranuleAndFiles = async (params: {
         deletedFiles: files,
       };
     } catch (error) {
```
Comment (jennyhliu) [inline packages/api/src/lib/granule-delete.ts:149]: When granule is not in dynamoDB, delete has error:
{ "error": { "name": "TypeError", "message": "Cannot read properties of undefined (reading 'files')", "stack": [ "TypeError: Cannot read properties of undefined (reading 'files')", " at deleteGranuleAndFiles (/var/task/webpack:/granule-delete.ts:148:119)", " at processTicksAndRejections (node:internal/process/task_queues:96:5)", " at del (/var/task/webpack:/endpoints/granules.js:551:27)" ] }, "level": "error", "message": "Cannot read properties of undefined (reading 'files')", "sender": "@cumulus/api", "timestamp": "2023-04-18T20:35:10.277Z" }

File: packages/api/lib/granule-remove-from-cmr.js [hunk 4]
```diff
@@ -29,15 +28,18 @@ const models = require('../models');
 const _removeGranuleFromCmr = async (granule, collectionId) => {
   log.info(`granules.removeGranuleFromCmrByGranule granule_id: ${granule.granule_id}, colletion_id: ${collectionId}`);
   if (!granule.published || !granule.cmr_link) {
-    throw new GranuleNotPublished(`Granule ${granule.granule_id} in Collection ${collectionId} is not published to CMR, so cannot be removed from CMR`);
+    log.warn(`Granule ${granule.granule_id} in Collection ${collectionId} is not published to CMR, so cannot be removed from CMR`);
```
Comment (botanical) [inline packages/api/lib/granule-remove-from-cmr.js:31]: 🙌

**Pull request merged**